src/web/rustdoc.rs

```diff
@@ -688,4 +700,90 @@ mod test {
             Ok(())
         });
     }
+
+    #[test]
+    fn base_redirect_handles_mismatched_separators() {
+        wrapper(|env| {
+            let db = env.db();
+
+            let rels = [
+                ("dummy-dash", "0.1.0"),
+                ("dummy-dash", "0.2.0"),
+                ("dummy_underscore", "0.1.0"),
+                ("dummy_underscore", "0.2.0"),
+                ("dummy_mixed-separators", "0.1.0"),
+                ("dummy_mixed-separators", "0.2.0"),
+            ];
+
+            for (name, version) in &rels {
+                db.fake_release()
+                    .name(name)
+                    .version(version)
+                    .rustdoc_file(&(name.replace("-", "_") + "/index.html"), b"")
+                    .create()?;
+            }
+
+            let web = env.frontend();
+
+            assert_redirect("/dummy_dash", "/dummy-dash/0.2.0/dummy_dash/", web)?;
+            assert_redirect("/dummy_dash/*", "/dummy-dash/0.2.0/dummy_dash/", web)?;
+            assert_redirect("/dummy_dash/0.1.0", "/dummy-dash/0.1.0/dummy_dash/", web)?;
+            assert_redirect(
+                "/dummy-underscore",
+                "/dummy_underscore/0.2.0/dummy_underscore/",
+                web,
+            )?;
+            assert_redirect(
+                "/dummy-underscore/*",
+                "/dummy_underscore/0.2.0/dummy_underscore/",
+                web,
+            )?;
+            assert_redirect(
+                "/dummy-underscore/0.1.0",
+                "/dummy_underscore/0.1.0/dummy_underscore/",
+                web,
+            )?;
+            assert_redirect(
+                "/dummy-mixed_separators",
+                "/dummy_mixed-separators/0.2.0/dummy_mixed_separators/",
+                web,
+            )?;
+            assert_redirect(
+                "/dummy_mixed_separators/*",
+                "/dummy_mixed-separators/0.2.0/dummy_mixed_separators/",
+                web,
+            )?;
+            assert_redirect(
+                "/dummy-mixed-separators/0.1.0",
+                "/dummy_mixed-separators/0.1.0/dummy_mixed_separators/",
+                web,
+            )?;
+
```

Comment:
Can you also add a few tests for `/crate/dummy_mixed-separators`?

Reply:
Added in the other test which checks that `/crate/:crate` paths are _not_ affected by this change.